### PR TITLE
ixing APPFAC-3372

### PR DESCRIPTION
**Pull request by apremalal**

### CODE REVIEW

File: modules/jaggery-apps/appmgt/src/site/pages/createapplication.jag [hunk 1]
```diff
@@ -22,6 +22,10 @@ var site = require("/site/conf/site.json");
 
 (function () {
     var type = request.getParameter('type');
+    var apptype = request.getParameter('apptype');
+    var apptypeIconColor = request.getParameter('apptypeIconColor');
+    var apptypeIconImageClass = request.getParameter('apptypeIconImageClass');
+    var appTypeDisplayName = request.getParameter('appTypeDisplayName');
```
Comment (hisamith) [inline modules/jaggery-apps/appmgt/src/site/pages/createapplication.jag]: This is wrong. We should not get apptypeIconColor ,apptypeIconImageClass ,appTypeDisplayName from the request. Those things should be taken from the apptype bean. 

https://wso2.org/jira/browse/APPFAC-3233

Comment (apremalal) [inline modules/jaggery-apps/appmgt/src/site/pages/createapplication.jag]: Let me have a look.

On Tue, Sep 15, 2015 at 10:44 PM, Samith Dassanayake <
notifications@github.com> wrote:

> In modules/jaggery-apps/appmgt/src/site/pages/createapplication.jag
> <https://github.com/wso2/product-af/pull/567#discussion_r39537509>:
>
> > @@ -22,6 +22,10 @@ var site = require("/site/conf/site.json");
> >
> >  (function () {
> >      var type = request.getParameter('type');
> > +    var apptype = request.getParameter('apptype');
> > +    var apptypeIconColor = request.getParameter('apptypeIconColor');
> > +    var apptypeIconImageClass = request.getParameter('apptypeIconImageClass');
> > +    var appTypeDisplayName = request.getParameter('appTypeDisplayName');
>
> This is wrong. We should not get apptypeIconColor ,apptypeIconImageClass
> ,appTypeDisplayName from the request. Those things should be taken from the
> apptype bean.
>
> https://wso2.org/jira/browse/APPFAC-3233
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wso2/product-af/pull/567/files#r39537509>.
>



-- 
*Anuruddha Premalal*
Software Eng. | WSO2 Inc.
Mobile : +94710461070
Web site : www.anuruddha.org

Comment (apremalal) [inline modules/jaggery-apps/appmgt/src/site/pages/createapplication.jag]: What is the rationale behind getting these information from appType bean?. Note that this is a 'page' and all the pages follow the request.getParameter way. Also I don't see any harm of using this way, particularly for this parameters come as POST not as url params.

Comment (hisamith) [inline modules/jaggery-apps/appmgt/src/site/pages/createapplication.jag]: Problem is whenever we redirected to this page, we have to set those parameters. I f we leave like this, when new element is introduced (per se "appTypeAnimation" :blush: ) we have to send it as this, when ever we call this page. So sending 'type' is enough  and all other things can be get from the type.

**Comment by apremalal** [comment]
Well if we consider the performance aspect reading it from the apptypeBean would be expensive unless we have cached the apptypeBeans. I'll look in to it.

**Pull request merged**